daemon/src/connection.rs

```diff
@@ -43,12 +44,6 @@ pub const MAX_RECONNECT_INTERVAL_SECONDS: u64 = 60;
 
 const TCP_TIMEOUT: Duration = Duration::from_secs(10);
 
-const HANDLE_CONTRACT_SETUP_TIMEOUT: Duration = Duration::from_secs(10);
```

Comment:
should this timeout be removed as a part of this PR? After all, you haven't converted the contract setup to send things in an async way.

Reply:
I re-introduced the timeouts for contract setup for now. Does not make it prettier, but yeah, that error path of getting stuck should be covered now.